src/test_workflow/integ_test/local_test_cluster_opensearch.py

```diff
@@ -0,0 +1,84 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import logging
+import os
+import subprocess
+import time
+
+import psutil  # type: ignore
+import requests
+import yaml
+
+from aws.s3_bucket import S3Bucket
+from manifests.bundle_manifest import BundleManifest
+from paths.tree_walker import walk
+from test_workflow.test_cluster import ClusterCreationException, TestCluster
+from test_workflow.test_recorder.test_recorder import TestRecorder
+from test_workflow.test_recorder.test_result_data import TestResultData
+from test_workflow.integ_test.service_opensearch import ServiceOpenSearch
+
+class LocalTestClusterOpenSearch(TestCluster):
+    """
+    Represents an on-box test cluster. This class downloads a bundle (from a BundleManifest) and runs it as a background process.
+    """
+    
+    def __init__(
+        self,
+        work_dir,
+        component_name,
+        additional_cluster_config,
+        bundle_manifest,
+        security_enabled,
+        component_test_config,
+        test_recorder: TestRecorder,
+        s3_bucket_name=None,
+    ):
+        self.manifest = bundle_manifest
+        self.work_dir = os.path.join(work_dir, "local-test-cluster")
```

Comment:
should we make `local-test-cluster`  a constant?

Reply:
Since this is not a blocking comment, I will address this in https://github.com/opensearch-project/opensearch-build/issues/838